ni/protobuf/types/waveform.proto

```diff
@@ -138,6 +138,30 @@ message WaveformAttributeValue
   }
 }
 
+// A digital waveform, which encapsulates digital data as integers, timing information, and extended properties.
+message DigitalWaveform
+{
+  // The time of the first sample in y_data.
+  PrecisionTimestamp t0 = 1;
+
+  // The time interval in seconds between data points in the waveform.
+  double dt = 2;
+
+  // The data values of the waveform.
+  repeated DigitalSample y_data = 3;
+
+  // The names and values of all waveform attributes.
+  // See the comment at near the top of this file for more details.
+  map<string, WaveformAttributeValue> attributes = 4;
+}
+
+// A digital sample, which consists of one or more digital signals.
+// We only need 8 bits to represent a digital state, but grpc doesn't define anything smaller than int32.
+// However, when serialized, these int32s will be optimized into a single byte each.
+message DigitalSample {
+   repeated int32 signals = 1;
```

Comment:
The optimal data type for an array of u8s is `bytes`.
https://protobuf.dev/programming-guides/proto3/#scalar